extension/content/components/recipes/form/arguments/ShowHeartBeat.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import {
+  HelpBlock,
+  FormGroup,
+  ControlLabel,
+  InputNumber,
+  Row,
+  Col,
+} from "rsuite";
+
+import {
+  ACTION_UPDATE_DATA,
+  useRecipeDetailsData,
+  useRecipeDetailsDispatch,
+} from "devtools/contexts/recipeDetails";
+
+import InputField from "devtools/components/recipes/form/arguments/fields/InputField";
+import SelectField from "devtools/components/recipes/form/arguments/fields/SelectField";
+import ToggleField from "devtools/components/recipes/form/arguments/fields/ToggleField";
+
+export default function ShowHeart() {
+  return (
+    <FormGroup>
+      <Row>
+        <Col xs={12}>
+          <InputField label="Survey ID" name="surveyID" />
+          <InputField label="Message" name="message" />
+          <InputField label="Learn More Message" name="learnMoreMessage" />
+          <RepeatOptionsField />
+        </Col>
+        <Col xs={12}>
+          <InputField
+            label="Engagement Button Label"
+            name="engagementButtonLabel"
+          />
+          <InputField label="Thanks Message" name="thanksMessage" />
+          <InputField label="Post-Answer URL" name="postAnswerUrl" />
+          <ToggleField
+            label="Include Telemetry UUID?"
+            name="includeTelemetryUUID"
+          >
+            Include UUID in Post-Answer URL and Telemetry.
+          </ToggleField>
+        </Col>
+      </Row>
+    </FormGroup>
+  );
+}
+
+function RepeatOptionsField() {
+  const SHOW_PROMPT_OPTIONS = [
+    { label: "Show this prompt once.", value: "once" },
+    {
+      label: "Show users every X days.",
+      value: "xdays",
+    },
+    {
+      label:
+        "Show until the user interacts with the heartbeat prompt, then never again.",
+      value: "nag",
+    },
+  ];
+
+  const repeatsEveryChangeSideEffect = ({ data, value }) => {
```

Comment:
nit: this should be `repeatOptionChangeSideEffect` since `repeatOption` is the field that is changing.